app/controllers/actions/NonIleQueryAction.scala

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.actions
+
+import config.AppConfig
+import controllers.exception.FeatureDisabledException
+import javax.inject.Inject
+import models.requests.AuthenticatedRequest
+import play.api.mvc._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class NonIleQueryAction @Inject()(appConfig: AppConfig)(implicit val exc: ExecutionContext)
```

Comment:
Bear in mind that this is another class that we'll need to remove once the feature flag is not needed.
An alternative could be to parameterize existing `IleQueryAction`

Reply:
Could do.   But if the parameter is a boolean it might get confusing as to which way it should be.  Again, classes are easy to delete when they're no longer needed.

Reply:
I think that this logic must stay after the feature flag is removed. I mean if we don't have feature flag user shouldn't be able to access this page.

Reply:
Once the feature flag is removed, we can delete the page that we are trying to protect (just like we are currently doing in the internal version of the service).

Reply:
Yeah, you're right